CMakeLists.txt

```diff
@@ -64,12 +64,21 @@ if(CMAKE_SYSTEM_NAME STREQUAL "Darwin" AND NOT CMAKE_HOST_APPLE)
   # So disable this tool check in further enable_language() commands.
   set(CMAKE_PLATFORM_HAS_INSTALLNAME FALSE)
 endif()
+
+list(APPEND CMAKE_MODULE_PATH ${PROJECT_SOURCE_DIR}/cmake/module)
+
+# Set the default config before enabling c++ so the compiler is never run with
+# the wrong build type's flags.
+include(DefaultConfig)
+set_default_config(RelWithDebInfo)
+
```

Comment:
Is the default build configuration really important for detecting the compiler and its capabilities?

Reply:
Not critically, no. I just figured it should for consistency. Is there a downside to this?